lib/internal/fs/streams.js

```diff
@@ -102,25 +107,22 @@ function ReadStream(path, options) {
     }
   }
 
+  this.on('open', () => {
+    // Do this in listener to maintain compat with e.g.
+    // graceful-fs.
+    this[kPending] = false;
```

Comment:
`this[kPending] = false` is not needed before `errorOrDestroy()`?

Reply:
no, I don't think so

Reply:
it exist to check whether 'open' has been emitted or not

Reply:
Sorry, I was wrong. It's needed before. I'm not quite sure what's the best way here...